packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -366,6 +369,19 @@ export class Dialog
   // #endregion
 
   // #region Private Methods
+
+  private handleFocusTrapDisabled(focusTrapDisabled: boolean): void {
+    if (!this.open) {
```

Comment:
shouldn't it deactivate if not open?

Reply:
The logic needs to handle changes to `open`, `modal` and `focusTrapDisabled`.

- if `open` is false => deactivateFocusTrap
- `modal` is true and `open` is true => activateFocusTrap
- if focusTrapDisabled is true and modal is false => deactivateFocusTrap

Reply:
I think this hasn't been resolved yet.

Reply:
I think the onOpen/onClose handles the deactivation part. We may not need this check at all. We can probably remove this function and call `activateFocusTrap` in its place

Reply:
Gotcha. Sounds like `focusTrapDisabled` is intended for use at "open" time. Maybe we can clarify this in the docs. cc @geospatialem @DitwanP

Reply:
Yeah, I think we handle that in multiple places. `onClose` takes care of `deactivateFocusTrap`, and the rest happens in 
` focusTrapDisabledOverride(): boolean {
    return !this.modal && this.focusTrapDisabled;
  } `